orttraining/orttraining/training_ops/cpu/quantization/fake_quant.cc

```diff
@@ -35,6 +36,13 @@ void FakeQuantPerTensor(OpKernelContext* ctx, const int64_t num_elements, const
         }
       });
 }
+
+template <typename T>
+void FakeQuantGradImpl(const Tensor& dY, const Tensor& gradient_mask, Tensor& dX) {
+  // If gradient_mask is true (i.e. quantization was in range), return dY, else return 0
+  MakeEigenArrayMap<T>(dX) = MakeEigenArrayMap<T>(dY) * MakeEigenArrayMap<bool>(gradient_mask).template cast<T>();
```

Comment:
If the compute here is so straightforward, is that better we use Mul node as the backward op. This may requires we output the masks to be in type T, do you think that would be a problem?

Reply:
We could do that. The only problem with that is that we would hold onto 4 times the memory as compared to the stashed boolean tensor. In fact, I would want to start using a bitset if possible (similar to dopout).